### PR TITLE
Handle globalPermission: allow image to be public

### PR DESCRIPTION
**Pull request by outscale-hmi**

### CODE REVIEW

**Comment by outscale-hmi** [comment]
Generate Test fail, which is normal : 
 echo "==> Checking that auto-generated code is not changed..."
            git diff --exit-code; if [ $? -eq 1 ]; then \
                echo "Found diffs in go generated code."; \
                echo "You can use the command: \`make generate\` to reformat code."; \
                exit 1; \

Adding GlobalPermission parameter in config change the generated file comparing to an old version (git diff fail)

**Comment by outscale-mdr** [comment]
@outscale-hmi , you can generate them and them in your PR

**Comment by outscale-hmi** [comment]
@outscale-mdr ti works now thanks.

File: builder/osc/common/step_update_omi_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+	StepUpdateOMIAttributes := StepUpdateOMIAttributes{
+		AccountIds:         []string{},
+		SnapshotAccountIds: []string{},
+		RawRegion:          "us-west-2",
+		GlobalPermission:   true,
+	}
+	state := tState()
+
+	StepUpdateOMIAttributes.Run(context.Background(), state)
```
Comment (outscale-mdr) [inline builder/osc/common/step_update_omi_test.go:39]: What does the test do ?

**Review by outscale-mdr — CHANGES_REQUESTED** [review]
Do you think you have to make some changes [there](https://github.com/outscale/packer-plugin-outscale/blob/649953fe86ca97eecf7e73af0109480ba2292705/builder/osc/common/step_update_omi.go#L28)  ?

File: builder/osc/common/step_update_omi.go [hunk 2]
```diff
@@ -58,26 +66,10 @@ func (s *StepUpdateOMIAttributes) Run(_ context.Context, state multistep.StateBa
...
-		// 		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
-		// 	},
-		// }
-
-		//regionconn := oapi.NewClient(newConfig, skipClient)
-
 		ui.Message(fmt.Sprintf("Updating: %s", omi))
 		updateImageRequest.ImageId = omi
-		_, _, err := regionconn.ImageApi.UpdateImage(context.Background(), &osc.UpdateImageOpts{
+		var err error
```
Comment (outscale-mgo) [inline builder/osc/common/step_update_omi.go:71]: I'm not sure this line (`var err error`) is useful

File: builder/osc/common/step_update_omi_test.go (hunk 1 above)
Comment (outscale-mgo) [reply builder/osc/common/step_update_omi_test.go:39]: yes, doesn't this test return an error if something (the StepUpdateOMIAttributes) doesn't work ?

**Review by outscale-mgo — CHANGES_REQUESTED** [review]

**Review by outscale-mgo — APPROVED** [review]

**Pull request merged**